Allow reading of KNMI hdf5 with read_hdf5

j-fannon opened the pull request:
Adding in some extra flags and decoding of KNMI hdf5 files e.g.:
3h acc: https://dataplatform.knmi.nl/dataset/radar-tar-corr-accum-03h-2-0
5min acc: https://dataplatform.knmi.nl/dataset/nl-rdr-data-rtcor-5m-tar-1-0

Requires the invert_data flag from @pollyaschm i.e. https://github.com/pollyaschm/harpIO/commit/57d51f3153babfa6323df51427e518b8b8f6c032.

Sample resulting plot for the 3h acc attached below. Tagging in @adeckmyn and @andrew-MET for review.

```
knmi_3h <- read_grid("RAD_NL25_RAC_03H_202507061200.h5",
                     "pcp",
                     file_format = "hdf5",
                     file_format_opts = harpIO::hdf5_opts(invert_data=F,
                                                           iflag = "knmi",
                                                           pcp_quant = "PRECIP_[MM]"))
```

<img width="619" height="647" alt="image" src="https://github.com/user-attachments/assets/1396e102-af19-43f6-bce2-0693e8641541" />

andrew-MET commented:
Tiny thing, as my initial reaction was that `hdf5_opts()` should handle this, and I see that it actually does. So, could you just edit the example to use `hdf5_opts()` rather than a raw list. Also, not quite as tiny(!), hdf5_opts() should be exported and documented with roxygen2 tags. 

@carlos9917 I think the invert_data argument deals with the issue you were having with DMI radar data. Can you confirm?

carlos9917 commented:
> Tiny thing, as my initial reaction was that `hdf5_opts()` should handle this, and I see that it actually does. So, could you just edit the example to use `hdf5_opts()` rather than a raw list. Also, not quite as tiny(!), `hdf5_opts()` should be exported and documented with roxygen2 tags.
> 
> @carlos9917 I think the invert_data argument deals with the issue you were having with DMI radar data. Can you confirm?

I use this and still see data inverted with DMI data

<img width="617" height="635" alt="image" src="https://github.com/user-attachments/assets/9fc86231-18ce-4b26-8e53-1becf19dc5b8" />
library(harp)
library(hdf5r)

obs_file_path <- "/media/cap/extra_work/verification/oper-harp-verif/ACCORD_VS_202507/sample_data/radar_dmi/sqpe/kavrrad_1h/2025/05"
![202505130600 kavrRAD 01](https://github.com/user-attachments/assets/5f72a46c-e105-4a13-a0e3-1f1f023232fb)


precip_ob <- harpIO::read_grid(
  paste0(obs_file_path,"/202505130600.kavrRAD.01.h5"),"Pcp",
  hdf5_opts = hdf5_opts(data_path = "/pcp/data1/data", odim = FALSE, meta = TRUE,invert_data=T)
)
plot_field(precip_ob)
File is attached (renamed as gif instead of h5 otherwise git will not take it)

carlos9917 commented:
![202505130600 kavrRAD 01](https://github.com/user-attachments/assets/9ddf0902-06af-47f5-9f27-176533c64acc)

andrew-MET commented:
Was it that you need `invert_data = FALSE` for your case?

j-fannon commented:
Thanks @andrew-MET - I've added some documentation and moved `hdf5_opts` to it's own function. The example is updated to reflect this. 

When writing the documentation I realised that not all options passed to `read_hdf5` are respected in a consistent way i.e. data_path and odim. Maybe correcting these should be a separate PR, however.

andrew-MET commented:
I'll merge this as it currently is in the next day or 2, so the other corrections can be done in a new PR
